### PR TITLE
Final pre-publish housekeeping

### PR DESCRIPTION
**Pull request by craiglabenz**
* Removed dependency on Flutter from packages - all are now pure-Dart
  * Various linting changes (removing redundant null-check operators) that emerged by moving from `flutter_lints` to `lints`
* Moved dependencies on `mediapipe_core` to `0.0.1` (already published)
* Added supported platforms declaration
* Added MediaPipe's license files

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]
A few more to fix:

```console
Analyzing mediapipe-core...

warning - lib/src/io/containers.dart:72:19 - The '!' will have no effect because the receiver can't be null. Try removing the '!' operator. - unnecessary_non_null_assertion
warning - lib/src/io/containers.dart:84:19 - The '!' will have no effect because the receiver can't be null. Try removing the '!' operator. - unnecessary_non_null_assertion
warning - lib/src/io/containers.dart:134:15 - The '!' will have no effect because the receiver can't be null. Try removing the '!' operator. - unnecessary_non_null_assertion
warning - lib/src/io/containers.dart:158:19 - The '!' will have no effect because the receiver can't be null. Try removing the '!' operator. - unnecessary_non_null_assertion
warning - lib/src/io/containers.dart:240:19 - The '!' will have no effect because the receiver can't be null. Try removing the '!' operator. - unnecessary_non_null_assertion
warning - lib/src/io/containers.dart:260:19 - The '!' will have no effect because the receiver can't be null. Try removing the '!' operator. - unnecessary_non_null_assertion
warning - lib/src/io/containers.dart:261:34 - The '!' will have no effect because the receiver can't be null. Try removing the '!' operator. - unnecessary_non_null_assertion
warning - lib/src/io/containers.dart:278:19 - The '!' will have no effect because the receiver can't be null. Try removing the '!' operator. - unnecessary_non_null_assertion
warning - lib/src/io/containers.dart:279:36 - The '!' will have no effect because the receiver can't be null. Try removing the '!' operator. - unnecessary_non_null_assertion
warning - lib/src/io/task_result.dart:41:15 - The '!' will have no effect because the receiver can't be null. Try removing the '!' operator. - unnecessary_non_null_assertion
warning - lib/src/io/task_result.dart:66:42 - The '!' will have no effect because the receiver can't be null. Try removing the '!' operator. - unnecessary_non_null_assertion
```

**Comment by craiglabenz** [comment]
As far as I can tell, these latest integration test failures are a false-positive. They passed until recently in CI and still pass locally.

File: packages/mediapipe-task-text/pubspec.yaml [hunk 1]
```diff
@@ -17,14 +25,13 @@ dependencies:
...
-  mediapipe_core:
-    path: ../mediapipe-core
+  mediapipe_core: ^0.0.1
   native_assets_cli: ^0.3.0
   native_toolchain_c: ^0.3.0
   path: ^1.8.3
 
 dev_dependencies:
   ffigen: ^9.0.1
-  flutter_lints: ^2.0.0
```
Comment (dcharkes) [inline packages/mediapipe-task-text/pubspec.yaml:28]: DBC: If I'm not mistaken the list of flutter lints is bigger than the lints package. So you might as well keep flutter_lints if you don't mind satisfying them. It makes the code more pleasant to read to typical flutter users.

**Comment by craiglabenz** [comment]
Merging as test failures are false positives from dart-lang/sdk#55712

**Pull request merged**